packages/odata-v4/src/uri-conversion/odata-uri.ts

```diff
@@ -31,5 +21,27 @@ function getExpandWrapped<
 export function createODataUri<DeSerializersT extends DeSerializers>(
   deSerializers: DeSerializersT
 ): ODataUri<DeSerializersT> {
-  return createODataUriBase(deSerializers, getExpandWrapped, getSelect);
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  function getExpandWrapped<EntityT extends Entity>(
+    selects: Selectable<EntityT, DeSerializersT>[],
+    expands: Expandable<
+      EntityT,
+      DeSerializersT,
+      EntityApi<EntityBase, DeSerializersT>
+    >[],
+    entityApi: EntityApi<Entity, DeSerializersT>
+  ): Partial<{ expand: string }> {
+    return getExpand(expands, entityApi);
+  }
+
+  // This enforces the same DeSerializersT on the getSelect method. If we change something on the selec method signature this will fail in contrast to use of as XYZ.
```

Comment:
I remember the context, but without this context I am not sure that I would know what this comment tells me.

```suggestion
  // This enforces the same DeSerializersT on the getSelect function. If we change something on the getSelect function signature this will fail in contrast to use of as XYZ.
```